File: output/ChatGPT/codenet/C++/Python/s324508229.py
n = int(input())
cnt = [[0] * 10 for _ in range(10)]

for i in range(1, n + 1):
    b = i % 10
    a = i
    while a >= 10:
        a //= 10
    cnt[a][b] += 1

ans = 0

for i in range(1, 10):
    for j in range(1, 10):
        ans += cnt[i][j] * cnt[j][i]

print(ans)

File: output/ChatGPT/codenet/C++/Java/s324508229.java
import java.util.Scanner;

public class s324508229 {
    public static void main(String[] args) {
        Scanner scanner = new Scanner(System.in);
        int n = scanner.nextInt();
        long[][] cnt = new long[10][10];
        for (int i = 0; i < 100; i++) {
            cnt[i % 10][i / 10] = 0;
        }
        for (int i = 1; i <= n; i++) {
            int b = i % 10;
            int a = i;
            while (a >= 10) {
                a /= 10;
            }
            cnt[a][b]++;
        }
        long ans = 0;
        for (int i = 1; i <= 9; i++) {
            for (int j = 1; j <= 9; j++) {
                ans += cnt[i][j] * cnt[j][i];
            }
        }
        System.out.println(ans);
    }
}